### PR TITLE
6 get comments by article

### PR DESCRIPTION
**Pull request by milbot1992**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -38,6 +38,62 @@ describe('GET /api', () => {
...
+                expect(article).toHaveProperty('topic')
+                expect(article).toHaveProperty('created_at')
+                expect(article).toHaveProperty('votes')
+                expect(article).toHaveProperty('article_img_url')
+                expect(article).toHaveProperty('comment_count')
+                expect(article).not.toHaveProperty('body')
+            })
+        })
+    })
+})
```
Comment (elijaharley) [inline __tests__/app.test.js:95]: Please see my comments on the previous pull request and refactor your tests accordingly to check key value pairs.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -38,6 +38,62 @@ describe('GET /api', () => {
...
+    })
+    test('should return 400 Bad Request if given an invalid id',()=>{
+        return request(app)
+        .get('/api/articles/notAnID/comments')
+        .expect(400)
+        .then(({body})=>{
+            expect(body.message).toBe('Invalid ID')
+        })
+    })
+})
```
Comment (elijaharley) [inline __tests__/app.test.js:139]: Good error handling tests here. Please also add a 200 test for a valid article with no comments which should return an empty array.

File: app/controllers/errors.controllers.js [hunk 3]
```diff
@@ -1,3 +1,17 @@
...
+exports.handleCustomErrors = (err, req, res, next) => {
+    if (err.status) {
+        res.status(err.status).send({ message: err.message })
+    }
+    next(err)
 }
+
+exports.handleServerErrors = (err, req, res, next) => {
+        res.status(500).send({message: 'Internal server error!'})
+}
```
Comment (elijaharley) [inline app/controllers/errors.controllers.js]: Nice extraction of error handlers here.

**Review by elijaharley — CHANGES_REQUESTED** [review]
Please review my comments from both pull requests and make the relevant changes before merging.
